# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/27 07:28:42-02:00 zaitcev@redhat.com 
#   [PATCH] USB: Prevent hiddev from looping
#   
#   In the loop, schedule() returns with the current state TASK_RUNNING, so at the
#   next revolution it returns immediately, and the task sits there burning CPU.
#   
#   The fix is to add the forgotten state assignment.
#   
#   Patch from David Micon.
# 
# drivers/usb/hiddev.c
#   2005/01/26 20:24:52-02:00 zaitcev@redhat.com +1 -0
#   USB: Prevent hiddev from looping
# 
diff -Nru a/drivers/usb/hiddev.c b/drivers/usb/hiddev.c
--- a/drivers/usb/hiddev.c	2005-02-03 05:58:42 -08:00
+++ b/drivers/usb/hiddev.c	2005-02-03 05:58:42 -08:00
@@ -328,6 +328,7 @@
 				}
 				
 				schedule();
+				set_current_state(TASK_INTERRUPTIBLE);
 			}
 
 			set_current_state(TASK_RUNNING);
